src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -419,11 +419,112 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_predicate_node(
+
+    // Case (i): Binary predicate with column = <subquery>. Equivalent to a semi-join with a table containing one row.
+    // We can get the statistics directly from the LQPSubqueryExpression.
+    if (predicate_condition == PredicateCondition::Equals) {
+      auto subquery_expression = std::dynamic_pointer_cast<LQPSubqueryExpression>(arguments[0]);
+      column_expression = arguments[1];
+
+      if (!subquery_expression) {
+        subquery_expression = std::dynamic_pointer_cast<LQPSubqueryExpression>(arguments[1]);
+        column_expression = arguments[0];
+      }
+
+      if (!subquery_expression || subquery_expression->is_correlated()) {
+        return input_table_statistics;
+      }
+
+      subquery_statistics = estimate_statistics(subquery_expression->lqp);
+    }
+
+    // Case (ii): Between predicate with column BETWEEN min(<subquery) AND max(<subquery>). Equivalent to a semi-join
+    // with the referenced table, where the min/max aggregates preserve the domain of the subquery result. However,
+    // we have to ensure that we have a min/max aggregate for the lower/upper bound on the same column.
+    if (predicate_condition == PredicateCondition::BetweenInclusive) {
+      column_expression = arguments[0];
+      const auto& lower_bound_subquery = std::dynamic_pointer_cast<LQPSubqueryExpression>(arguments[1]);
+      const auto& upper_bound_subquery = std::dynamic_pointer_cast<LQPSubqueryExpression>(arguments[2]);
+      if (!lower_bound_subquery || !upper_bound_subquery || lower_bound_subquery->is_correlated() ||
+          upper_bound_subquery->is_correlated()) {
+        return input_table_statistics;
+      }
+
+      // Check that input nodes provide only a single AggregateExpression.
+      const auto& lower_bound_lqp = *lower_bound_subquery->lqp;
+      const auto& upper_bound_lqp = *upper_bound_subquery->lqp;
+      const auto& lower_bound_node_expressions = lower_bound_lqp.node_expressions;
+      const auto& upper_bound_node_expressions = upper_bound_lqp.node_expressions;
+      if (lower_bound_node_expressions.size() != 1 || upper_bound_node_expressions.size() != 1) {
+        return input_table_statistics;
+      }
+      const auto& lower_bound_aggregate_expression =
+          std::dynamic_pointer_cast<AggregateExpression>(lower_bound_node_expressions.front());
+      const auto& upper_bound_aggregate_expression =
+          std::dynamic_pointer_cast<AggregateExpression>(upper_bound_node_expressions.front());
+      if (!lower_bound_aggregate_expression || !upper_bound_aggregate_expression) {
+        return input_table_statistics;
+      }
+
+      // Check that the AggregateFunctions are as expected and are performed on the same column, and the nodes have the
+      // same input.
+      auto subquery_origin_node = lower_bound_lqp.left_input();
+      if (lower_bound_aggregate_expression->aggregate_function != AggregateFunction::Min ||
```

Comment:
That's pretty restrictive (I would formulate a `between max() and max()`. I guess it's like that because we haven't encoutered other cases? If so, add a comment?

Reply:
Of course you can, but then the predicate is not equivalent to a semin-join with the input node of the aggregates.